cogito/protocol.go

```diff
@@ -215,15 +219,37 @@ func (src *Source) Validate() error {
 	// Validate mandatory fields.
 	//
 	var mandatory []string
-	if src.Owner == "" {
-		mandatory = append(mandatory, "owner")
+
+	// This is the Source validation about custom sinks,
+	// the second argument represents the put params that is nil
+	// at this stage.
+	_, err := MergeAndValidateSinks(src.Sinks, nil)
```

Comment:
I really don't like the idea of merging the sinks, throwing the result away and then  restart from scratch with the argument of the merge.
If this newly introduced function returned the merged sinks as a `Set` instead of a list of strings, you could use it straight away.
```diff
-	_, err := MergeAndValidateSinks(src.Sinks, nil)
+	sinks, err := MergeAndValidateSinks(src.Sinks, nil)
	if err != nil {
		return fmt.Errorf("source: invalid sink(s): %w", err)
	}

-	sinks := sets.From(src.Sinks...)
	if sinks.Size() == 0 || sinks.Contains("github") {
```

Reply:
Not exactly, this is the source validation checking what is mandatory and what is optional.
We are not interested to get the sinks (defaults or overridden), so the first step is to check if there is any error giving `src.Sinks` as argument. I don't want to have all the supported sinks if there is nothing configured, that's why I'm throwing away `_` the sinks returned by the function, but only any eventual error.

Reply:
You are interested in using what you validated and you return what you validated. So you it.

However, your comment made me realize that the code can be cleaned up further when using the returned sinks. The `if sinks.Size() == 0 || sinks.Contains("github") {` no longer needs to check if it's empty because what was returned is always a non-empty set of sinks. So you can just do `if sinks.Contains("github") {`. That's an excellent cleanup as it removes the needs to know about the default's logic here too.

Reply:
> You are interested in using what you validated and you return what you validated. So you it.
> 
> However, your comment made me realize that the code can be cleaned up further when using the returned sinks. The `if sinks.Size() == 0 || sinks.Contains("github") {` no longer needs to check if it's empty because what was returned is always a non-empty set of sinks. So you can just do `if sinks.Contains("github") {`. That's an excellent cleanup as it removes the needs to know about the default's logic here too.

ah, I get your point, let me do a try.

Reply:
I'm sorry it cannot work. The scope of `Validate()`, it's just to raise earlier if the resource is misconfigured (missing mandatory parameter. If I get all the default sinks at this stage, all the parameters become mandatory (I've 27 tests failing). 
We should consider to refactor this logic, but it's outside this PR. 
Also `Validate()` runs for the `check` and the `get` , there are some units tests around this.